src/hexkit/correlation.py

```diff
@@ -80,6 +84,16 @@ async def set_correlation_id(correlation_id: str):
         yield
 
 
+@asynccontextmanager
+async def set_new_correlation_id():
+    """Set a new correlation ID for the life of the context."""
+    correlation_id = new_correlation_id()
+
+    async with set_context_var(correlation_id_var, correlation_id):
+        log.info("Set context correlation ID to %s", correlation_id)
+        yield
```

Comment:
If you changed this to `yield correlation_id`, would that allow you to update tests to reference the new value directly instead of making a subsequent call to `get_correlation_id()`? So in tests it would then be:
```python
async with set_new_correlation_id() as temp_correlation_id:
    assert temp_correlation_id != correlation_id
```

Reply:
Good idea, implemented in 0a28adb now.